sdks/opik_optimizer/src/opik_optimizer/base_optimizer.py

```diff
@@ -272,6 +275,79 @@ def _serialize_tools(prompt: "chat_prompt.ChatPrompt") -> list[dict[str, Any]]:
                     serialized_tools.append({k: v for k, v in tool.items() if k})
             return serialized_tools
 
+    # ------------------------------------------------------------------
```

Comment:
BaseOptimizer is already overloaded and not easy to track. Maybe worth encapsulating the random sampling logic to a separate class with a compact API rather than continuing to grow the base class.